rest_framework/exceptions.py

```diff
@@ -65,15 +73,61 @@ def __str__(self):
 # from rest_framework import serializers
 # raise serializers.ValidationError('Value was invalid')
 
+ErrorDetails = namedtuple('ErrorDetails', ['message', 'code'])
+
+
 class ValidationError(APIException):
     status_code = status.HTTP_400_BAD_REQUEST
+    code = None
 
-    def __init__(self, detail):
-        # For validation errors the 'detail' key is always required.
-        # The details should always be coerced to a list if not already.
-        if not isinstance(detail, dict) and not isinstance(detail, list):
-            detail = [detail]
-        self.detail = _force_text_recursive(detail)
+    def __init__(self, detail, code=None):
+        if code:
+            self.full_details = ErrorDetails(detail, code)
+        else:
+            self.full_details = detail
```

Comment:
I'm not sure to follow.

It seems to me that this is what I'm doing right now:
> The population of .full_details. Should just be storing the message and code at this point.

It does store message and code for single items, a list when a list is provided and a dict when a dict is provided
Basically it does store what is passed to the `__init__` method.

We then need to transform whatever is given as the `detail` parameter to the former `self.detail` format which is what this code does (quite painfully).